ecosystem/sep-0038.md

```diff
@@ -0,0 +1,176 @@
+## Preamble
+
+```
+SEP: 0038
+Title: Request for Quote API
+Author: Jake Urban <@jakeurban>
+Track: Standard
+Status: Draft
+Created: 2021-04-09
+Discussion: https://github.com/stellar/stellar-protocol/issues/901
+Version 1.0.0
+```
+
+## Summary
+
+This protocol enables anchors to accept off-chain assets in exchange for different on-chain assets, and vice versa. Specifically, it enables anchors to provide quotes that can referenced within the context of existing Stellar Ecosystem Proposals. How the exchange of assets is facilitated is outside the scope of this document.
+
+## Motivation
+
+Anchoring an asset and issuing an asset are distinct functions that have different buisness and technical requirements. However, issuing an asset has often been a prerequisite for anchoring an asset. This protocol enables anchors to transfer value on and off the Stellar network for their clients regardless of whether a one-for-one reserve-backed Stellar asset existed for the off-chain asset held by the anchor's clients.
+
+Removing this requirement for anchors also provides downstream benefits to ecosystem participants generally. Enabling anchors to accept any Stellar asset will naturally decrease liquidity fragmentation on the decentralized exchange, leading to greater market depth and tigher spreads between trading pairs.
+
+## Specification
+
+### Authentication
+
+All endpoints require authentication in the form of a [SEP-10](sep-0010.md) JSON Web Token (JWT) in the `Authorization` header of the request. Authenticated accounts must exist on the network in order to make requests to `POST /quote`, as described in the [Protecting Against Bad Actors](#protecting-against-bad-actors) section.
+
+### Content Type
+
+All endpoints accept in requests the following `Content-Type`:
+- `application/json`
+
+All endpoints respond with content type:
+- `application/json`
+
+### Endpoints
+
+* [`GET /info`](#get-info)
+* [`GET /prices`](#get-prices)
+* [`GET /price`](#get-price)
+* [`POST /quote`](#post-quote)
+
+### GET Info
+
+This endpoint describes the supported Stellar assets and off-chain currencies available for trading. Note that the anchor may not support a trading pair between every Stellar asset and off-chain currency listed. Use the [`GET /prices`](#get-prices) endpoint to see which pairs are supported.
+
+#### Request
+
+No request arguments required.
+
+#### Response
+
+Name | Type | Description
+-----|------|------------
+`assets` | array | An array of objects describing the Stellar assets available in exchange for one or more of the `currencies` listed.
+`currencies` | array | An array of objects describing the off-chain currencies available in exchange for one or more of the `assets` listed.
```

Comment:
❓ Can `currencies` be non-currencies, such as Bitcoin? The distinction seems to be on-ledger assets and off-ledger assets. Thoughts on if there's a better term than `currencies`?

`assets` => `stellar-assets`
`currencies` => `external-assets`?

Reply:
I was also thinking this when drafting it. I didn't love the names I thought of but thats not a reason to implicitly exclude a set of assets that could be used here.

I think `stellar` and `external` are good prefixes, although I think we should stick with underscores to match our existing APIs.

Reply:
💯 on underscores. Not sure what I was thinking there.

Reply:
Another thought is could we distinguish stellar assets from external assets by their value/identity rather than the field. I'm thinking of a company who provides an API across any number of network's. There might be existing standards we can buy into such that we could say something like base `stellar:USDC:circle.com` counter `ethereum:...`. there was some discussion about namespacing like this on the SEP-32 discussion thread, so I think this is a thing. Maybe we could leverage it.

Reply:
This was the discussion: https://groups.google.com./g/stellar-dev/c/rWThI_BIs0Y/m/H9EBrlsHBwAJ

Reply:
I like the idea of namespacing assets by provider (or network). The existing cross-chain API I looked at isn't built for off-chain assets, which is requirement for us. I like the syntax suggested in the google group thread:

```
namespace:symbol:[domain or 'native']
```

For now we don't need to define `tranche` or `instance` but we could extend the convention in the future.

For now we can document the convention in this protocol but I could see it becoming its own SEP.

Reply:
We could start with only defining `stellar:` and `iso4217:` for now, and expand as needed. I don't think we need an exhaustive list.